core/src/test/java/org/apache/accumulo/core/data/KeyTest.java

```diff
@@ -79,6 +79,144 @@ public void testCopyData() {
     assertEquals(k1, k2);
 
   }
+  
+  @Test
+  public void testCopyDataWithByteArrayConstructors() {
+    byte[] row = "r".getBytes();
+    byte[] cf = "cf".getBytes();
+    byte[] cq = "cq".getBytes();
+    byte[] cv = "cv".getBytes();
+
+    Key kRow = new Key(row);
+    Key kRowcolFam = new Key(row, cf);
+    Key kRowcolFamColQual = new Key(row, cf, cq);
+    Key kRowcolFamColQualColVis = new Key(row, cf, cq, cv);
+    Key kRowcolFamColQualColVisTimeStamp = new Key(row, cf, cq, cv, 5L);
+
+    // test row constructor
+    assertNotSame(row, kRow.getRowBytes());
+    assertNotSame(cf, kRow.getColFamily());
+    assertNotSame(cq, kRow.getColQualifier());
+    assertNotSame(cv, kRow.getColVisibility());
+
+    assertNotSame(row, kRow.getRowData().getBackingArray());
+    assertNotSame(cf, kRow.getColumnFamilyData().getBackingArray());
+    assertNotSame(cq, kRow.getColumnQualifierData().getBackingArray());
+    assertNotSame(cv, kRow.getColumnVisibilityData().getBackingArray());
+    
+    // test row, column family constructor
+    assertNotSame(row, kRowcolFam.getRowBytes());
+    assertNotSame(cf, kRowcolFam.getColFamily());
+    assertNotSame(cq, kRowcolFam.getColQualifier());
+    assertNotSame(cv, kRowcolFam.getColVisibility());
+
+    assertNotSame(row, kRowcolFam.getRowData().getBackingArray());
+    assertNotSame(cf, kRowcolFam.getColumnFamilyData().getBackingArray());
+    assertNotSame(cq, kRowcolFam.getColumnQualifierData().getBackingArray());
+    assertNotSame(cv, kRowcolFam.getColumnVisibilityData().getBackingArray());
+    
+    // test row, column family, column qualifier constructor
+    assertNotSame(row, kRowcolFamColQual.getRowBytes());
+    assertNotSame(cf, kRowcolFamColQual.getColFamily());
+    assertNotSame(cq, kRowcolFamColQual.getColQualifier());
+    assertNotSame(cv, kRowcolFamColQual.getColVisibility());
+
+    assertNotSame(row, kRowcolFamColQual.getRowData().getBackingArray());
+    assertNotSame(cf, kRowcolFamColQual.getColumnFamilyData().getBackingArray());
+    assertNotSame(cq, kRowcolFamColQual.getColumnQualifierData().getBackingArray());
+    assertNotSame(cv, kRowcolFamColQual.getColumnVisibilityData().getBackingArray());
+    
+    // test row, column family, column qualifier, column visibility constructor
+    assertNotSame(row, kRowcolFamColQualColVis.getRowBytes());
+    assertNotSame(cf, kRowcolFamColQualColVis.getColFamily());
```

Comment:
Not sure, but I think different types are being compared (byte[] and Text)?  If so seems like they could never be the same.

Reply:
the Text objects are covered to byte[] before getting stored...if I am reading the code correctly. II added an equals method for the byte[] to make sure